### PR TITLE
stacks-js-rs: Serialize fee wallet commands.

### PR DESCRIPTION
**Pull request by sergey-shandar**
This PR serialize all required commands to `stacks.ts` script but it doesn't properly call `stacks.js` package functions yet.

#84

### CODE REVIEW

File: stacks-js-rs/tests/rpc.rs [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
+    Ok(x)
+}
 
 fn test_wrap() -> Result<(), Error> {
-    let mut js = Js::new(".")?;
+    let mut js = Js::new("./mirror.ts")?;
     {
-        let result = js.call(from_str("{\"b\":[],\"a\":2}")?)?;
-        assert_eq!(result.to_string(), "[\"object\",{\"a\":2,\"b\":[]}]");
+        let result = js.call::<_, Value>(&to_value("{\"b\":[],\"a\":2}")?)?;
```
Comment (netrome) [inline stacks-js-rs/tests/rpc.rs:24]: `Value` here is a very vague type name without the namespace information. Can you alias it as `JsonValue` in the use statement?

**Review by netrome — CHANGES_REQUESTED** [review]
Overall looks good to me, but I'd request to reduce the amount of data we serialize and send to JS before merging this.

File: stacks-js-rs/stacks.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+type Mint = {
+    readonly amount: number
+    readonly block_height: number
+    readonly burn_header_hash: BurnchainHeaderHash
+    readonly memo: readonly number[]
+    readonly peg_wallet_address: PoxAddress
+    readonly recipient: PrincipalData
+    readonly txid: string
+    readonly vtxindex: number
```
[the remark this answers is not in the document]
Comment (netrome) [reply stacks-js-rs/stacks.ts:50]: Edit: Looking at the testnet [sBTC contract](https://github.com/stacks-network/stacks-blockchain/pull/3499/files#diff-42dbc9fe309ad9c61d62ed3518345dcf477e2bba101408726aea8c3f8683a1df), the `txid` is also important here.

[the remark this answers is not in the document]
Comment (igorsyl) [reply stacks-js-rs/stacks.ts:50]: FYI @netrome source of truth for sBTC contract is [here](https://github.com/Trust-Machines/core-eng/blob/main/sbtc-ops/clarinet/contracts/sbtc-alpha.clar).

[the remark this answers is not in the document]
Comment (netrome) [reply stacks-js-rs/stacks.ts:50]: Right, thanks that's better. The method signature is the same, so `txid`, `recipient` and `amount` should be enough here. We probably also need to send the coordinator credentials in some format to sign the contract call transactions.

File: stacks-js-rs/stacks.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    readonly block_height: number
+    readonly burn_header_hash: readonly number[]
+    readonly fulfillment_fee: number
+    readonly memo: readonly number[]
+    readonly peg_wallet_address: PoxAddress
+    readonly recipient: PoxAddress
+    readonly signature: string
+    readonly txid: string
+    readonly vtxindex: number
+}
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply stacks-js-rs/stacks.ts]: We can restructure internal RPC call as we want, may be in the next PR.

**Review by netrome — APPROVED** [review]

**Review by xoloki — APPROVED** [review]
Lgtm, didn't set any obvious problems in the rust code, but I'm not really qualified to judge the js or ts

**Comment by sergey-shandar** [comment]
> Lgtm, didn't set any obvious problems in the rust code, but I'm not really qualified to judge the js or ts

I hope, we will get rid of the JS in one of the future PR and replace it with Rust code.

**Review by donpdonp — APPROVED** [review]
.rs traits - great, is actually a start on stacks.rs with a paticular (deno) implementation.
.js files - had a call with sergey as to why this layer is needed
stacks.js/deno - I'd wager stack.rs implementation is equal to the overhead of this approach.

the code here looks great for what its supposed to do.

**Comment by igorsyl** [comment]
> .rs traits - great, is actually a start on stacks.rs with a paticular (deno) implementation. .js files - had a call with sergey as to why this layer is needed stacks.js/deno - I'd wager stack.rs implementation is equal to the overhead of this approach.
> 
> the code here looks great for what its supposed to do.

I love the push back @donpdonp ! Let's keep watching over each other!

**Pull request merged**